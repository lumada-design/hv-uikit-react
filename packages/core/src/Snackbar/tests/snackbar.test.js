/* eslint-env jest */

import React from "react";
import { mount } from "enzyme";
import "jest-canvas-mock";
import { Snackbar as MaterialSnackbar } from "@mui/material";

import { Add } from "@hitachivantara/uikit-react-icons";
import { HvButton, HvProvider, HvSnackbar, HvSnackbarContent } from "../..";

describe("Snackbar ", () => {
  const wrapper = mount(
    <HvProvider cssBaseline="none">
      <HvSnackbar id="Snackbar" />
    </HvProvider>
  );

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper.find(HvSnackbar)).toMatchSnapshot();
  });

  it("should render the Snackbar component", () => {
    const component = wrapper.find(HvSnackbar);
    expect(component.length).toBe(1);
  });

  it("shouldn't render the SnackbarComponent component as the Snackbar isn't open", () => {
    const component = wrapper.find(HvSnackbarContent);
    expect(component.length).toBe(0);
  });

  it("should render the SnackbarComponent component as the Snackbar is open", () => {
    const component = mount(
      <HvProvider cssBaseline="none">
        <HvSnackbar open transitionDirection="right" />
      </HvProvider>
    ).find(HvSnackbarContent);
    expect(component.length).toBe(1);
  });

  it("shouldn't render icon when default", () => {
    const component = mount(
      <HvProvider cssBaseline="none">
        <HvSnackbar open showIcon transitionDirection="up" />
      </HvProvider>
    )
      .find(HvSnackbarContent)
      .find("svg");

    expect(component.length).toBe(0);
  });

  it("shouldn't render the success icon", () => {
    const component = mount(
      <HvProvider cssBaseline="none">
        <HvSnackbar open variant="success" showIcon={false} transitionDirection="down" />
      </HvProvider>
    )
      .find(HvSnackbarContent)
      .find("svg");

    expect(component.length).toBe(0);
  });

  it("should render the success icon", () => {
    const component = mount(
      <HvProvider cssBaseline="none">
        <HvSnackbar open variant="success" showIcon />
      </HvProvider>
    )
      .find(HvSnackbarContent)
      .find("svg");

    expect(component.length).toBe(1);
  });

  it("should render the warning icon", () => {
    const component = mount(
      <HvProvider cssBaseline="none">
        <HvSnackbar open variant="warning" showIcon />
      </HvProvider>
    )
      .find(HvSnackbarContent)
      .find("svg");

    expect(component.length).toBe(1);
  });

  it("should render the error icon", () => {
    const component = mount(
      <HvProvider cssBaseline="none">
        <HvSnackbar open variant="error" showIcon />
      </HvProvider>
    )
      .find(HvSnackbarContent)
      .find("svg");

    expect(component.length).toBe(1);
  });

  it("should render the custom icon", () => {
    const component = mount(
      <HvProvider cssBaseline="none">
        <HvSnackbar open customIcon={<Add />} />
      </HvProvider>
    )
      .find(HvSnackbarContent)
      .find("svg");

    expect(component.length).toBe(1);
  });

  it("should render the action when a component is passed", () => {
    const component = mount(
      <HvProvider cssBaseline="none">
        <HvSnackbar open variant="success" action={<a href=" ">Event</a>} />
      </HvProvider>
    )
      .find(HvSnackbarContent)
      .find("a");

    expect(component.length).toBe(1);
  });

  it("should render with the correct offset", () => {
    const offset = 10;
    let component = mount(
      <HvProvider cssBaseline="none">
        <HvSnackbar open offset={offset} />
      </HvProvider>
    ).find(MaterialSnackbar);

    expect(component.get(0).props.style).toEqual({ top: `${offset}px` });

    component = mount(
      <HvProvider cssBaseline="none">
        <HvSnackbar
          open
          offset={offset}
          anchorOrigin={{
            vertical: "bottom",
            horizontal: "right",
          }}
        />
      </HvProvider>
    ).find(MaterialSnackbar);

    expect(component.get(0).props.style).toEqual({ bottom: `${offset}px` });
  });

  it("should render the action when a structure is passed", () => {
    const component = mount(
      <HvProvider cssBaseline="none">
        <HvSnackbar
          id="Snackbar"
          open
          variant="success"
          action={{
            id: "testButton",
            label: "test",
          }}
        />
      </HvProvider>
    )
      .find(HvSnackbarContent)
      .find(HvButton);

    expect(component.length).toBe(1);
    expect(component.prop("category")).toBe("semantic");
  });
});
