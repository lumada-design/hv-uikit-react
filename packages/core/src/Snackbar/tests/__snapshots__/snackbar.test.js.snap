// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar  should render correctly 1`] = `
<WithStyles(HvSnackbar)
  id="Snackbar"
>
  <HvSnackbar
    classes={
      Object {
        "anchorOriginBottomCenter": "HvSnackbar-anchorOriginBottomCenter",
        "anchorOriginBottomLeft": "HvSnackbar-anchorOriginBottomLeft",
        "anchorOriginBottomRight": "HvSnackbar-anchorOriginBottomRight",
        "anchorOriginTopCenter": "HvSnackbar-anchorOriginTopCenter",
        "anchorOriginTopLeft": "HvSnackbar-anchorOriginTopLeft",
        "anchorOriginTopRight": "HvSnackbar-anchorOriginTopRight",
      }
    }
    id="Snackbar"
  >
    <ForwardRef(Snackbar)
      TransitionComponent={[Function]}
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      autoHideDuration={5000}
      classes={
        Object {
          "anchorOriginBottomCenter": "HvSnackbar-anchorOriginBottomCenter",
          "anchorOriginBottomLeft": "HvSnackbar-anchorOriginBottomLeft",
          "anchorOriginBottomRight": "HvSnackbar-anchorOriginBottomRight",
          "anchorOriginTopCenter": "HvSnackbar-anchorOriginTopCenter",
          "anchorOriginTopLeft": "HvSnackbar-anchorOriginTopLeft",
          "anchorOriginTopRight": "HvSnackbar-anchorOriginTopRight",
        }
      }
      id="Snackbar"
      open={false}
      style={
        Object {
          "top": "60px",
        }
      }
      transitionDuration={300}
    />
  </HvSnackbar>
</WithStyles(HvSnackbar)>
`;
